crypto/addresses/src/bech32.rs

```diff
@@ -0,0 +1,195 @@
+use crate::{Address, AddressError, Prefix};
+
+const CHARSET: &[u8] = b"qpzry9x8gf2tvdw0s3jn54khce6mua7l";
+const REV_CHARSET: [u8; 123] = [
+    100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100,
+    100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100,
+    100, 100, 15, 100, 10, 17, 21, 20, 26, 30, 7, 5, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100,
+    100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100,
+    100, 100, 100, 29, 100, 24, 13, 25, 9, 8, 23, 100, 18, 22, 31, 27, 19, 100, 1, 0, 3, 16, 11, 28, 12, 14, 6, 4, 2,
+];
+
+// Checksome for bech32
+// https://bch.info/en/specifications
+fn polymod<'data, I>(values: I) -> u64
+where
+    I: Iterator<Item = &'data u8>,
+{
+    let mut c = 1u64;
+    for d in values {
+        let c0 = c >> 35;
+        c = ((c & 0x07ffffffff) << 5) ^ (*d as u64);
+
+        if c0 & 0x01 != 0 {
+            c ^= 0x98f2bc8e61;
+        }
+        if c0 & 0x02 != 0 {
+            c ^= 0x79b76d99e2;
+        }
+        if c0 & 0x04 != 0 {
+            c ^= 0xf33e5fb3c4;
+        }
+        if c0 & 0x08 != 0 {
+            c ^= 0xae2eabe2a8;
+        }
+        if c0 & 0x10 != 0 {
+            c ^= 0x1e4f43e470;
+        }
+    }
+    c ^ 1
+}
+
+fn checksum(payload: &[u8], prefix: &[u8]) -> u64 {
+    polymod(
+        prefix
+            .iter()
+            .chain(&[0u8])
+            .chain(payload)
+            .chain(&vec![0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8]),
```

Comment:
You can remove the `vec!` and just pass an array of zeros

Reply:
just `[0u8; 8]`

Reply:
If i need to pass it as cloned inter, should I use `[0u8; 8].iter().cloned()`? It seems that just passing `[0u8; 8]` translate to `Iterator<Item=&u8>`

Reply:
you should add `.copied()` after the first `iter()` and then `[0u8; 8]` should work as is:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3b92190d8ae19cce7b56140e2806f0d4

Reply:
done